sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CollationExpressionSuite.scala

```diff
@@ -174,10 +174,10 @@ class CollationExpressionSuite extends SparkFunSuite with ExpressionEvalHelper {
       ("aa", "UTF8_BINARY_LCASE", UTF8String.fromString("aa").getBytes),
       ("AA", "UTF8_BINARY_LCASE", UTF8String.fromString("aa").getBytes),
       ("aA", "UTF8_BINARY_LCASE", UTF8String.fromString("aa").getBytes),
-      ("", "UNICODE", UTF8String.fromString("").getBytes),
-      ("aa", "UNICODE", UTF8String.fromString("aa").getBytes),
-      ("AA", "UNICODE", UTF8String.fromString("AA").getBytes),
-      ("aA", "UNICODE", UTF8String.fromString("aA").getBytes),
+      ("", "UNICODE", Array[Byte](1, 1, 0)),
+      ("aa", "UNICODE", Array[Byte](42, 42, 1, 6, 1, 6, 0)),
+      ("AA", "UNICODE", Array[Byte](42, 42, 1, 6, 1, -36, -36, 0)),
+      ("aA", "UNICODE", Array[Byte](42, 42, 1, 6, 1, -59, -36, 0)),
```

Comment:
nit: I can see that you have been using the same format for UNICODE_CI, so it makes sense to use the same representation. However, I think it would be more easy to follow if a hex representation was used instead of the integer representation (especially for the negative values, I think it is more intuitive to write them in hex form, especially given the fact that we are talking about bytes)